### PR TITLE
NEW REST API improvements

### PR DESCRIPTION
**Pull request by jtraulle**
New REST API Explorer branding and improvements

branding

- Header of REST API Explorer has been Dolibarr colored
- API Methods are now sorted alphabetically
- API endpoints with same names are now sorted in this order (GET, POST, PUT, PATCH, DELETE)

----

New REST API improvements to thirdparties bank accounts and gateways

gateways

- API endpoints has been added and documented to allow user to add, update and delete payment gateways like the one used by Stripe module (stored in table **llx_societe_account**)
- API endpoints for **CompanyBankAccounts** has been slightly modified to be harmonized with other API endpoints (this should not be a problem since these methods are not yet in Dolibarr latest stable version (aka 8.0))
- Some documentation has been improved (to be shown in REST explorer)
- `societeaccount.class.php` has been slightly modified to include missing `$login` member ; `notnull` value of member `fk_website` is now `-1` to put it to `NULL` when updating the record.

### CODE REVIEW

File: htdocs/societe/class/api_thirdparties.class.php [hunk 1]
```diff
@@ -114,8 +115,7 @@ function get($id)
 	 * @param   string  $sqlfilters Other criteria to filter answers separated by a comma. Syntax example "(t.nom:like:'TheCompany%') and (t.date_creation:<:'20160101')"
 	 * @return  array               Array of thirdparty objects
 	 */
-    function index($sortfield = "t.rowid", $sortorder = 'ASC', $limit = 100, $page = 0, $mode=0, $sqlfilters = '')
-    {
+	function index($sortfield = "t.rowid", $sortorder = 'ASC', $limit = 100, $page = 0, $mode=0, $sqlfilters = '') {
```
Comment (frederic34) [inline htdocs/societe/class/api_thirdparties.class.php]: please keep opening brace on new line, i have uniformised all the others

Comment (jtraulle) [reply htdocs/societe/class/api_thirdparties.class.php]: Thanks @frederic34, sorry about that :confused: 
Must be my IDE ; I have fixed it now :wink:

File: htdocs/societe/class/api_thirdparties.class.php [hunk 2]
```diff
@@ -1151,40 +1147,398 @@ function updateCompanyBankAccount($socid, $id, $request_data = null)
...
+	 * @param int $id ID of thirdparty
+	 * @param int $bankaccount_id ID of CompanyBankAccount
 	 *
 	 * @return int -1 if error 1 if correct deletion
 	 *
-	 * @url DELETE {socid}/CompanyBankAccount/{id}
+	 * @url DELETE {id}/bankaccounts/{bankaccount_id}
 	 */
-    function deleteCompanyBankAccount($id, $socid)
+	function deleteCompanyBankAccount($bankaccount_id, $id)
```
Comment (frederic34) [inline htdocs/societe/class/api_thirdparties.class.php]: inverted params?

Comment (jtraulle) [reply htdocs/societe/class/api_thirdparties.class.php]: No, I have left the same order (which is determined by the order of parameters in the @url annotation).
I have just changed the name of the parameters but yes, maybe I should switch the parameter order in the function signature. It does not change anything but maybe it is better for clarity :wink: 

Tell me what you think :slightly_smiling_face:

Comment (frederic34) [reply htdocs/societe/class/api_thirdparties.class.php]: ah ok, so you have to invert the @param comment to have the same order than in the function

Comment (jtraulle) [reply htdocs/societe/class/api_thirdparties.class.php]: I have switched function signature parameters. That way, @param, @url annotations and function signature are in the same order. Make more sense :wink:

**Comment by jtraulle** [comment]
Because this PR slightly modify #9016, I prefer to ping @yolocodefrench to let him/her know :wink:

File: htdocs/societe/class/api_thirdparties.class.php [hunk 3]
```diff
@@ -1109,33 +1106,32 @@ function createCompanyBankAccount($socid, $request_data = null)
...
 		}
 
 		$account = new CompanyBankAccount($this->db);
 
-		$account->fetch($id, $socid, -1, '');
+		$account->fetchFromApi($bankaccount_id, $id);
 
 
-		if($account->socid != $socid){
+		if($account->socid != $bankaccount_id){
```
Comment (frederic34) [inline htdocs/societe/class/api_thirdparties.class.php]: $socid become $id

Comment (jtraulle) [reply htdocs/societe/class/api_thirdparties.class.php]: Thanks :wink:

**Pull request merged**